File: apps/mo2-android/android/core/model/src/main/kotlin/jp/developershub/dub/mo2/core/model/Mobile.kt
// Mirror of @dub/types `mobile` namespace (packages/types/src/mobile.ts).
// MO3 mobile-bff owns this namespace; it is the OpenAPI generation source.
// Sync/mutation shapes are STUB in the frozen contract and omitted here until
// MO3's later wave lands them.
package jp.developershub.dub.mo2.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class MobilePlatform {
    @SerialName("ios") IOS,
    @SerialName("android") ANDROID,
}

@Serializable
data class MobileHomeResponse(
    val upcomingEvents: List<EventSummary>,
    val myTasks: List<TaskSummary>,
    val unreadCount: Int,
)

@Serializable
data class MobileEventOverviewResponse(
    val event: EventSummary,
    val capabilities: List<PermissionKey>,
)

@Serializable
data class RegisterDeviceRequest(
    val platform: MobilePlatform,
    val pushToken: String,
)

@Serializable
data class RegisterDeviceResponse(
    val deviceId: String,
)

@Serializable
data class DeviceDto(
    val id: String,
    val platform: MobilePlatform,
    val registeredAt: ISODateTime,
)

@Serializable
data class MobilePushPayload(
    val title: String,
    val body: String,
    val data: Map<String, String>? = null,
)

File: packages/types/src/mobile.ts
// mobile — MO3 mobile-bff namespace (owner=MO3). OpenAPI generation source
// (swift5/kotlin). Sync/mutations are STUB (implemented in a later wave).
import type { UserId, ISODateTime, Paginated, CursorQuery } from "./common";
import type { EventSummary } from "./event";
import type { TaskSummary } from "./task";
import type { PermissionKey } from "./identity";
import type { NotificationType } from "./notification";

export type MobilePlatform = "ios" | "android";

export interface MobileHomeResponse {
  upcomingEvents: EventSummary[];
  myTasks: TaskSummary[];
  unreadCount: number;
}
export interface MobileEventOverviewResponse {
  event: EventSummary;
  capabilities: PermissionKey[];
}
export interface RegisterDeviceRequest {
  platform: MobilePlatform;
  pushToken: string;
}
export interface RegisterDeviceResponse {
  deviceId: string;
}
export interface DeviceDto {
  id: string;
  platform: MobilePlatform;
  registeredAt: ISODateTime;
}
export interface PushDispatchRequest {
  userId: UserId;
  type: NotificationType;
  payload: MobilePushPayload;
}
export interface MobilePushPayload {
  title: string;
  body: string;
  data?: Record<string, string>;
}

// ---- STUB: sync/mutations は実装後段で確定 ----
export interface SyncQuery extends CursorQuery {
  since?: ISODateTime; // STUB
}
export interface SyncResponse extends Paginated<unknown> {
  serverTime: ISODateTime; // STUB
}
